sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java

```diff
@@ -420,7 +421,7 @@ private static Tag tag(long id) {
 
     private void throwIfClosed() {
         if (this.closed.get()) {
-            throw new TransportException(lenientFormat("%s is closed", this), null);
+            throw new ClosedClientTransportException(lenientFormat("%s is closed", this), null);
```

Comment:
Curious: why not directly throw a InternalServerError here?

Reply:
My reasoning is this - we do a wrapping of `InternalServerErrorException` in the outer layers such as `StoreReader` - so wanted to throw `ClosedClientTransportException` instead to keep the changes isolated. Also, I make a decision to re-throw (in `ConsistencyReader`) if I know the first-level inner type is `ClosedClientTransportException`.

Reply:
Revisiting this - throwing `InternalServerErrorException` from `RntbdTransportClient` directly.